app/src/org/commcare/activities/connect/ConnectActivity.java

```diff
@@ -69,22 +89,82 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
                     .setPopUpTo(navController.getGraph().getStartDestinationId(), true)
                     .build();
             navController.navigate(fragmentId, bundle, options);
+        } else {
```

Comment:
Nit: this can just be:
} else if (redirectionAction != null) {

Reply:
Done